Dockerfile

```diff
@@ -13,7 +13,7 @@ WORKDIR /build
 # install dependencies
 RUN python -m venv /opt/venv && \
   . /opt/venv/bin/activate && \
-  pip install --no-cache-dir -U 'pip<20' && \
+  pip install --no-cache-dir -U pip && \
```

Comment:
The default `pip` version is too old - and led to the error [here](https://github.com/RasaHQ/rasa/pull/9436/checks?check_run_id=3422475326)

Alternatively we can also pin the `pip` version? Doing `<20` doesn't make sense anymore in my opinion though.

```
EnvCommandError
222
#15 41.34 
223
#15 41.34   Command ['/opt/venv/bin/pip', 'install', '--no-deps', '/root/.cache/pypoetry/artifacts/8a/22/bd/d3a3de944e9ba43058166f3b10a0cc8ad8d71470238f2b9b9825e552d3/cryptography-3.4.8-cp36-abi3-manylinux_2_24_x86_64.whl'] errored with the following return code 1, and output: 
224
#15 41.34   ERROR: cryptography-3.4.8-cp36-abi3-manylinux_2_24_x86_64.whl is not a supported wheel on this platform.
225
#15 41.34   
```

Reply:
pinned it to the major as a compromise between playing it safe and maintainability.